File: backend/src/main/java/com/pgvaale/backend/entity/ServiceProvider.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "service_providers")
public class ServiceProvider {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String type; // TIFFIN or MAID
    private String region;
    private boolean approved = false;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getRegion() { return region; }
    public void setRegion(String region) { this.region = region; }
    public boolean isApproved() { return approved; }
    public void setApproved(boolean approved) { this.approved = approved; }
} 

File: PGVaaleDotNetBackend/Entities/ServiceProvider.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class ServiceProvider
    {
        [Key]
        public long Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Type { get; set; } = string.Empty; // TIFFIN or MAID

        public string Region { get; set; } = string.Empty;

        public bool Approved { get; set; } = false;
    }
}
